ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -0,0 +1,706 @@
+            for each detected date, and second list containing corresponding original substrings in text
+
+        Examples:
+            obj = DateDetector('date')
+            text = '16th august to 27th august'
+            obj.detect_entity(text)
+
+                Output:
+                    [{'detection_method': 'message',
+                          'from': True,
+                          'normal': False,
+                          'start_range': False,
+                          'end_range': False
+                          'text': '16th august',
+                          'to': False,
+                          'value': {'dd': 16, 'mm': 8, 'type': 'date', 'yy': 2017}},
+
+                    {'detection_method': 'message',
+                          'from': False,
+                          'normal': False,
+                          'start_range': False,
+                          'end_range': False
+                          'text': '27th august',
+                          'to': True,
+                          'value': {'dd': 27, 'mm': 8, 'type': 'date', 'yy': 2017}}]
+
+        Additionally this function assigns these lists to self.date and self.original_date_text attributes
+        respectively.
+        """
+        self.text = ' ' + text.lower() + ' '
+        self.processed_text = self.text
+        self.tagged_text = self.text
+        date_data = []
+        if run_model:
+            date_data = self._date_model_detection()
+        if not run_model or not date_data:
+            date_data = self._detect_date()
+        return date_data
+
+    def _detect_date(self):
+        """
+        Detects a date and categorises it into "from", "to", "start_range", "end_range" and "normal" attributes
+
+        Returns:
+            It returns the list of dictionary containing the fields like detection_method, from, normal, to,
+            text, value, start_range, end_range
+        """
+        final_date_dict_list = []
+        date_dict_list = self._detect_range()
+        final_date_dict_list.extend(date_dict_list)
+        self._update_processed_text(date_dict_list)
+        date_dict_list = self._detect_return_date()
+        final_date_dict_list.extend(date_dict_list)
+        self._update_processed_text(date_dict_list)
+        date_dict_list = self._detect_departure_date()
+        final_date_dict_list.extend(date_dict_list)
+        self._update_processed_text(date_dict_list)
+        date_dict_list = self._detect_any_date()
+        final_date_dict_list.extend(date_dict_list)
+        self._update_processed_text(date_dict_list)
+
+        return final_date_dict_list
+
+    def _detect_range(self):
+        """
+        Finds <any text><space(s)><'-' or 'to'><space(s)><any text> in the given text.
+        It  splits the text into two parts on '-' or 'to'
+        and detects the start range in the first (left) part and detects end range in the second (right) part
```

Comment:
This functionality is not yet supported for other languages. So please add that comment or check if it is possible to add it in this release.

Reply:
done